Common/src/main/java/vazkii/botania/data/recipes/RecipeProvider.java

```diff
@@ -876,22 +876,29 @@ private void registerMisc(Consumer<FinishedRecipe> consumer) {
 
 		ShapedRecipeBuilder.shaped(ModBlocks.defaultAltar)
 				.define('P', ModTags.Items.PETALS)
-				.define('S', Items.COBBLESTONE_SLAB)
 				.define('C', Items.COBBLESTONE)
-				.pattern("SPS")
+				.pattern("CPC")
+				.pattern(" C ")
+				.pattern("CCC")
+				.unlockedBy("has_item", conditionsFromTag(ModTags.Items.PETALS))
+				.save(consumer);
+		ShapedRecipeBuilder.shaped(ModBlocks.mossyAltar)
+				.define('P', ModTags.Items.PETALS)
+				.define('C', Items.MOSSY_COBBLESTONE)
+				.pattern("CPC")
 				.pattern(" C ")
 				.pattern("CCC")
 				.unlockedBy("has_item", conditionsFromTag(ModTags.Items.PETALS))
 				.save(consumer);
 		for (String metamorphicVariant : LibBlockNames.METAMORPHIC_VARIANTS) {
-			Block altar = Registry.BLOCK.getOptional(prefix("apothecary_" + metamorphicVariant.replaceAll("_", ""))).get();
```

Comment:
Why was this `.replaceAll("_", "")` even here? None of the variant names have `_` and i don't see why you'd strip them anyway

Reply:
Looks like @Hubry made it like that when setting up the datagen two years ago, doesn't seem like there was a particularly good reason for it

Reply:
Might have been just copypasting from an another spot? I have no idea anymore